### PR TITLE
[exporter/splunkhecexporter] Add `OtelAttrsToHec` struct and related config fields

### LINKED ISSUE #35092: `hec_metadata_to_otel_attrs` option is confusing on exporter/splunkhec
### Component(s)

exporter/splunkhec

### Describe the issue you're reporting

The config option `hec_metadata_to_otel_attrs` is confusing when used with the splunkhec _exporter_ as the actual effect is the opposite, i.e. taking otel attributes and using them as HEC metadata when sending to some destination.

https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/bd66cebb0a7848c5e38a8d29a26532ee2c5a1d91/exporter/splunkhecexporter/config.go#L120

### PR DESCRIPTION
**Pull request by bderrly**
**Description:** Adding a new config fields `otel_attrs_to_hec_metadata/*` to replace the `hec_metadata_to_otel_attrs/*` config fields.

**Link to tracking Issue:** fixes #35092

**Testing:** No new tests added, modified existing tests.

**Documentation:** Adding field-level description to exporter README.md.

### CODE REVIEW

File: exporter/splunkhecexporter/README.md [hunk 1]
```diff
@@ -54,10 +54,18 @@ The following configuration options can also be configured:
...
 - `export_raw` (default = false): send only the log's body, targeting a Splunk HEC raw endpoint.
+- `otel_attrs_to_hec_metadata/source` (default = 'com.splunk.source'): Specifies the mapping of a specific unified model attribute value to the standard source field of a HEC event.
+- `otel_attrs_to_hec_metadata/sourcetype` (default = 'com.splunk.sourcetype'): Specifies the mapping of a specific unified model attribute value to the standard sourcetype field of a HEC event.
+- `otel_attrs_to_hec_metadata/index` (default = 'com.splunk.index'):  Specifies the mapping of a specific unified model attribute value to the standard index field of a HEC event.
+- `otel_attrs_to_hec_metadata/host` (default = 'host.name'):  Specifies the mapping of a specific unified model attribute value to the standard host field and the `host.name` field of a HEC event.
 - `hec_metadata_to_otel_attrs/source` (default = 'com.splunk.source'): Specifies the mapping of a specific unified model attribute value to the standard source field of a HEC event.
+  **Deprecated** (v0.109.0): prefer `otel_attrs_to_hec_metadata/source`.
```
Comment (atoulme) [inline exporter/splunkhecexporter/README.md:62]:
```suggestion
  **Deprecated** (v0.111.0): prefer `otel_attrs_to_hec_metadata/source`.
```

File: exporter/splunkhecexporter/config.go [hunk 2]
```diff
@@ -116,8 +116,14 @@ type Config struct {
 
 	// App version is used to track telemetry information for Splunk App's using HEC by App version. Defaults to the current OpenTelemetry Collector Contrib build version.
 	SplunkAppVersion string `mapstructure:"splunk_app_version"`
+
+	// OtelAttrsToHec creates a mapping from attributes to HEC specific metadata: source, sourcetype, index and host.
+	OtelAttrsToHec splunk.HecToOtelAttrs `mapstructure:"otel_attrs_to_hec_metadata"`
+
 	// HecToOtelAttrs creates a mapping from attributes to HEC specific metadata: source, sourcetype, index and host.
+	// Deprecated: [v0.109.0] Use OtelAttrsToHec instead.
```
Comment (atoulme) [inline exporter/splunkhecexporter/config.go:124]:
```suggestion
	// Deprecated: [v0.111.0] Use OtelAttrsToHec instead.
```

File: exporter/splunkhecexporter/config.go [hunk 3]
```diff
@@ -116,8 +116,14 @@ type Config struct {
...
 	HecToOtelAttrs splunk.HecToOtelAttrs `mapstructure:"hec_metadata_to_otel_attrs"`
+
```
Comment (atoulme) [inline exporter/splunkhecexporter/config.go:126]:
```suggestion
```

File: exporter/splunkhecexporter/logdata_to_splunk.go [hunk 4]
```diff
@@ -23,22 +23,53 @@ const (
 	traceIDFieldKey = "trace_id"
 )
 
+// copyOtelAttrs copies values from HecToOtelAttrs to OtelAttrsToHec struct.
+func copyOtelAttrs(config *Config) {
```
Comment (atoulme) [inline exporter/splunkhecexporter/logdata_to_splunk.go:27]: Please add tests

File: exporter/splunkhecexporter/config.go (hunk 3 above)
Comment (bderrly) [reply exporter/splunkhecexporter/config.go]: Done

File: exporter/splunkhecexporter/config.go (hunk 2 above)
Comment (bderrly) [reply exporter/splunkhecexporter/config.go]: Done

File: exporter/splunkhecexporter/README.md (hunk 1 above)
Comment (bderrly) [reply exporter/splunkhecexporter/README.md]: Done

File: exporter/splunkhecexporter/logdata_to_splunk.go (hunk 4 above)
Comment (bderrly) [reply exporter/splunkhecexporter/logdata_to_splunk.go:27]: Done

**Review by atoulme — APPROVED** [review]

**Comment by bderrly** [comment]
I updated the deprecation target versions. Do I need approval from @dmitryax or can @atoulme merge this?

**Comment by bderrly** [comment]
I would still like to see this PR merge. Is there some step that I am overlooking on how to get this to be merged?

**Comment by bderrly** [comment]
@atoulme, is there some way to revive this PR or do I need to create another one?

File: exporter/splunkhecexporter/README.md [hunk 5]
```diff
@@ -54,10 +54,18 @@ The following configuration options can also be configured:
...
 - `export_raw` (default = false): send only the log's body, targeting a Splunk HEC raw endpoint.
+- `otel_attrs_to_hec_metadata/source` (default = 'com.splunk.source'): Specifies the mapping of a specific unified model attribute value to the standard source field of a HEC event.
+- `otel_attrs_to_hec_metadata/sourcetype` (default = 'com.splunk.sourcetype'): Specifies the mapping of a specific unified model attribute value to the standard sourcetype field of a HEC event.
+- `otel_attrs_to_hec_metadata/index` (default = 'com.splunk.index'):  Specifies the mapping of a specific unified model attribute value to the standard index field of a HEC event.
+- `otel_attrs_to_hec_metadata/host` (default = 'host.name'):  Specifies the mapping of a specific unified model attribute value to the standard host field and the `host.name` field of a HEC event.
 - `hec_metadata_to_otel_attrs/source` (default = 'com.splunk.source'): Specifies the mapping of a specific unified model attribute value to the standard source field of a HEC event.
+  **Deprecated** (v0.113.0): prefer `otel_attrs_to_hec_metadata/source`.
```
Comment (atoulme) [inline exporter/splunkhecexporter/README.md:62]: need to update the version numbers now

**Comment by atoulme** [comment]
Changes still look good to me. You need to update the README for the deprecation notice.

File: exporter/splunkhecexporter/README.md (hunk 5 above)
Comment (bderrly) [reply exporter/splunkhecexporter/README.md]: Done.

**Review by atoulme — APPROVED** [review]

**Comment by Bre77** [comment]
Did https://help.splunk.com/en/splunk-observability-cloud/manage-data/splunk-distribution-of-the-opentelemetry-collector/get-started-with-the-splunk-distribution-of-the-opentelemetry-collector/collector-components/exporters/splunk-hec-exporter#name-otel_to_hec_fields not get updated to reflect this change?

**Pull request merged**